src/cypress-api/support.ts

```diff
@@ -0,0 +1,90 @@
+  // @ts-expect-error will fix when Cypress has its own package
+  cy.wrap({}).as('archive');
+  // @ts-expect-error will fix when Cypress has its own package
+  cy.wrap([]).as('manualSnapshots');
+
+  // since we don't know where the user will navigate, we'll archive whatever domain they're on first.
+  // @ts-expect-error will fix when Cypress has its own package
+  cy.intercept(`**/*`, (req) => {
+    // don't archive the page itself -- we'll do that with rrweb
+    // TODO: See if this will work for both slash and not slash endings or if we have to do same "first URL visited" stuff
+    if (!pageUrl) {
+      pageUrl = new URL(req.url);
+      return;
+    }
+
+    const url = new URL(req.url);
+    if (url.origin !== pageUrl.origin) {
+      return;
+    }
+
+    // cached (304) responses don't provide a body, so we need to make sure cache is blown away
+    // (https://glebbahmutov.com/blog/cypress-intercept-problems/#cached-response)
+    // https://github.com/cypress-io/cypress/issues/15680
+    delete req.headers['if-none-match'];
+    // I added this since some css files still are cached... not sure if this is great
+    // (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/304)
+    delete req.headers['if-modified-since'];
+    // @ts-expect-error will fix when Cypress has its own package
+    req.continue((response) => {
+      // @ts-expect-error will fix when Cypress has its own package
+      cy.get('@archive')
+        .then((archive: ResourceArchive) => {
+          const updatedArchive = {
+            ...archive,
+            [response.url]: {
+              statusCode: response.statusCode,
+              statusText: response.statusMessage,
+              body: response.body,
+            },
+          };
+          return updatedArchive;
+        })
+        .as('archive');
+    });
+  });
+};
+
+const completeArchive = () => {
+  // @ts-expect-error will fix when Cypress has its own package
+  cy.get('@archive').then((archive) => {
+    // can we be sure this always fires after all the requests are back?
+    // @ts-expect-error will fix when Cypress has its own package
+    cy.document().then((doc) => {
+      const snap = snapshot(doc, { noAbsolute: true });
+      // @ts-expect-error will fix when Cypress has its own package
+      cy.get('@manualSnapshots').then((manualSnapshots = []) => {
+        // pass the snapshot to the server to write to disk
+        // @ts-expect-error will fix when Cypress has its own package
+        cy.task('archiveCypress', {
+          // @ts-expect-error will fix when Cypress has its own package
+          testTitle: Cypress.currentTest.title,
+          domSnapshots: [...manualSnapshots, snap],
+          resourceArchive: archive,
+          chromaticStorybookParams: {
+            // @ts-expect-error will fix when Cypress has its own package
+            diffThreshold: Cypress.env('diffThreshold'),
+          },
+        });
+      });
+    });
```

Comment:
Because this runs in the `afterEach` hook of the test, I'm going off the assumption that all network requests have finished. If so, this will be much simpler than our [Playwright version](https://github.com/chromaui/test-archiver/blob/main/src/resource-archive/index.ts#L56-L82). But let me know if I may be mistaken in my assumption.

Reply:
We should be able to test it out (with a page that has some slow resolving requests). I would expect if your test body doesn't wait on the page it would complete without resolving a requests though.